AGENTS.md

```diff
@@ -0,0 +1,443 @@
+
+#### Option 2: Azure OpenAI (requires Azure resources and costs)
+
+1. Install Azure Developer CLI: https://aka.ms/install-azd
+2. Login to Azure:
+   ```bash
+   azd auth login
+   # Or for Codespaces: azd auth login --use-device-code
+   ```
+3. Provision resources (creates resource group, OpenAI account, deploys gpt-4o and text-embedding-3-small):
+   ```bash
+   azd provision
+   ```
+   This will:
+   - Prompt for environment name (e.g., "agents-demos")
+   - Prompt for subscription
+   - Prompt for location
+   - Create Azure resources using `infra/main.bicep`
+   - Automatically run `infra/write_dot_env.sh` (or `.ps1` on Windows) to create `.env`
+
+4. After provisioning, the `.env` file will be created automatically with all needed variables.
+
+5. To clean up resources:
+   ```bash
+   azd down
+   ```
+
+#### Option 3: OpenAI.com (requires API key and costs)
+
+```bash
+cp .env.sample.openai .env
+# Edit .env and set:
+# API_HOST=openai
+# OPENAI_API_KEY=your_openai_api_key
+# OPENAI_MODEL=gpt-4o-mini
+```
+
+#### Option 4: Ollama (requires local Ollama installation)
+
+1. Install Ollama: https://ollama.com/
+2. Pull a model:
+   ```bash
+   ollama pull llama3.1
+   ```
+3. Create .env:
+   ```bash
+   cp .env.sample.ollama .env
+   # Edit .env and set:
+   # API_HOST=ollama
+   # OLLAMA_ENDPOINT=http://localhost:11434/v1
+   # OLLAMA_MODEL=llama3.1
+   ```
+   **Important:** If running in a Dev Container, use `http://host.docker.internal:11434/v1` instead of `localhost`.
+
+### Running Scripts
+
+After environment setup and .env configuration, run any script:
+
+```bash
+python chat.py
+python function_calling_basic.py
+python rag_csv.py  # Requires requirements-rag.txt to be installed
+```
+
+For Spanish versions:
+```bash
+python spanish/chat.py
+```
+
+## Running Tests
+
+### Linting with Ruff
+
+```bash
+# After installing requirements-dev.txt
+ruff check .
+```
+
+To auto-fix issues:
+```bash
+ruff check . --fix
+```
+
+### Formatting with Black
+
+Check formatting:
+```bash
+black . --check --verbose
+```
+
+Auto-format:
+```bash
+black .
+```
+
+### Pre-commit Hooks
+
+Install pre-commit hooks to automatically run checks before commits:
+```bash
+pre-commit install
+```
+
+Run manually:
+```bash
+pre-commit run --all-files
+```
+
+### Integration Tests
+
+The repository has limited automated testing via GitHub Actions. The primary test runs basic scripts with GitHub Models:
+
+```bash
+# This is what the CI does (requires GitHub token):
+export API_HOST=github
+export GITHUB_TOKEN=$YOUR_TOKEN
+export GITHUB_MODEL=openai/gpt-4o-mini
+python chat.py
+python spanish/chat.py
+```
+
+**Note:** Most scripts are demonstration scripts, not unit-tested. Changes to scripts should be manually verified by running them.
+
+## Upgrading Python Dependencies
```

Comment:
This repo does not actually compile dependencies, so this sectio can be entirely removed.